### ISSUE TITLE
Error on Alpine Linux 3.9 and lower: smbc_setOptionProtocols: symbol not found

### ISSUE THREAD

**Issue by mlocati** [eduardok/libsmbclient-php#79]
Start a Docker container with Alpine Linux 3.9, for example with:

```sh
docker run --rm -it php:7.3-cli-alpine3.9 sh
```

In that docker image, type these commands:

```sh
apk add $PHPIZE_DEPS libsmbclient samba-dev
pecl install smbclient
docker-php-ext-enable smbclient
php --ri smbclient
```

We have this error (which occurs whenever we run php):

```
PHP Warning:  PHP Startup: Unable to load dynamic library 'smbclient.so' (tried: /usr/local/lib/php/extensions/no-debug-non-zts-20180731/smbclient.so (Error relocating /usr/local/lib/php/extensions/no-debug-non-zts-20180731/smbclient.so: smbc_setOptionProtocols: symbol not found), /usr/local/lib/php/extensions/no-debug-non-zts-20180731/smbclient.so.so (Error loading shared library /usr/local/lib/php/extensions/no-debug-non-zts-20180731/smbclient.so.so: No such file or directory)) in Unknown on line 0
Extension 'smbclient' not present.
```

This error occurs with smbclient version 1.0.2 and 1.0.3. With previous versions everything works fine.
For example, installing version 1.0.1:

```sh
apk add $PHPIZE_DEPS libsmbclient samba-dev
pecl install smbclient-1.0.1
docker-php-ext-enable smbclient
php --ri smbclient
```

outputs:

```
smbclient

smbclient Support => enabled
smbclient extension Version => 1.0.1
libsmbclient library Version => 4.8.12
```

**Comment by remicollet** [comment]
Use a real distro ;)
or try PR #80

**Comment by remicollet** [comment]
Notice, there is no benefit to use 1.0.2 or 1.0.3 if you don't have smbc_setOptionProtocols

**Comment by mlocati** [comment]
> Use a real distro ;)

What do you mean? Are you telling that Alpine Linux is not a real distro?? Have you ever tried it??? It's sooooooo fast and tiny, it's wonderful for docker images :wink:

> there is no benefit to use 1.0.2 or 1.0.3 if you don't have smbc_setOptionProtocols

Well, I know, but the current behavior is a bit strange to me.
IMHO, we have two alternatives:

- require a recent version of the libsmbclient (and compilation should fail if the version is too old)
- add some `#ifdef` to the C code so that we call libsmbclient functions only if present in the current version of libsmbclient

Furthermore, if you'll release version 1.1.0 of the smbclient PHP package with super fancy new features, people using old libsmbclient versions are forced to still use version 1.0.1.

**Comment by remicollet** [comment]
> IMHO, we have two alternatives:
> 
>     * require a recent version of the libsmbclient (and compilation should fail if the version is too old)

Seems bad (for ex, RHEL 7 or 8 have old libsmbclient, but this function is available)

>     * add some `#ifdef` to the C code so that we call libsmbclient functions only if present in the current version of libsmbclient

What is done in PR #80

**Comment by eduardok** [comment]
Fine, released 1.0.4.
Thanks @remicollet
